codegen/smithy-aws-swift-codegen/src/main/kotlin/software/amazon/smithy/aws/swift/codegen/customization/RulesBasedAuthSchemeResolverGenerator.kt

```diff
@@ -125,6 +128,34 @@ class RulesBasedAuthSchemeResolverGenerator {
                         )
                         write("validAuthOptions.append(sigV4Option)")
                         dedent()
+                        // sigv4-s3express case
+                        if (ctx.service.isS3) {
+                            write("case .sigV4S3Express(let param):")
+                            indent()
+                            write(
+                                "var authOption = \$N(schemeID: \$S)",
+                                SmithyHTTPAuthAPITypes.AuthOption,
+                                "aws.auth#sigv4-s3express",
+                            )
+                            write(
+                                "authOption.signingProperties.set(key: \$N.signingName, value: param.signingName)",
+                                SmithyHTTPAuthAPITypes.SigningPropertyKeys,
+                            )
+                            write(
+                                "authOption.signingProperties.set(key: \$N.signingRegion, value: param.signingRegion)",
+                                SmithyHTTPAuthAPITypes.SigningPropertyKeys,
+                            )
+                            write(
+                                "authOption.identityProperties.set(key: \$N.bucket, value: serviceParams.bucket)",
+                                AWSSDKIdentityTypes.AWSIdentityPropertyKeys,
+                            )
+                            write(
+                                "authOption.identityProperties.set(key: \$N.s3ExpressClient, value: S3ExpressCreateSessionClient())",
+                                AWSSDKIdentityTypes.AWSIdentityPropertyKeys,
+                            )
+                            write("validAuthOptions.append(authOption)")
+                            dedent()
+                        }
```

Comment:
In the S3 rules-based auth scheme resolver only, the S3 Express auth option is added.

(See the generated code above for more discussion of the content here.)